packages/core-utils/src/itinerary.ts

```diff
@@ -99,11 +99,11 @@ export function legDropoffRequiresAdvanceBooking(leg: Leg): boolean {
  */
 export function isFlex(leg: Leg): boolean {
   return (
-    isReservationRequired(leg) ||
-    isCoordinationRequired(leg) ||
-    legDropoffRequiresAdvanceBooking(leg) ||
-    isAdvanceBookingRequired(leg?.pickupBookingInfo) ||
-    legContainsGeometry(leg)
+    leg?.stopCalls?.some(call =>
+      // Flex calls are "Location" or "LocationGroup"
+      // eslint-disable-next-line no-underscore-dangle
+      call?.stopLocation.__typename.includes("Location")
```

Comment:
`stopLocation` is declared optional/nullable in the types but is assumed defined here. Should it be non-optional/non-nullable in the type declaration too?

Reply:
Replace `.includes` with `.startsWith`.

Reply:
Great catches!